keps/sig-node/2371-cri-pod-container-stats/README.md

```diff
@@ -0,0 +1,762 @@
++    UInt64Value inodes = 8;
+=}
+```
+
+All together, below is the proposition for the new `ContainerStats` object:
+```
+=// ContainerStats provides the resource usage statistics for a container.
+=message ContainerStats {
+=    // Information of the container.
++    // Corresponds to the Stats Summary API ContainerStats Name field
+=    ContainerAttributes attributes = 1;
+=    // CPU usage gathered from the container.
++    // Corresponds to Stats Summary API ContainerStats CPUStats field
+=    CpuUsage cpu = 2;
+=    // Memory usage gathered from the container.
++    // Corresponds to Stats Summary API ContainerStats MemoryStats field
+=    MemoryUsage memory = 3;
+=    // Usage of the writable layer.
++    // Corresponds to Stats Summary API ContainerStats Rootfs field
+=    FilesystemUsage writable_layer = 4;
++    // Stats pertaining to container logs usage of filesystem resources
++    // Logs.UsedBytes is the number of bytes used for the container logs.
++    FilesystemUsage logs = 5;
+=}
+```
+**TODO**: in Stats Summary API ContainerStats object, there's a time field. Do we want to add this to `ContainerStats`?
+	- Advantages to doing so are: the CRI implementation doesn't need to scrape the stats everytime the CRI call happens (more flexibility)
+
+Note the omission of the Stats Summary API ContainerStats Accelerators field. With its [deprecation](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/1867-disable-accelerator-usage-metrics/README.md), it was deemed not worth implementing in the CRI.
+
+[summary-container-stats-object]: https://github.com/kubernetes/kubernetes/blob/release-1.20/staging/src/k8s.io/kubelet/pkg/apis/stats/v1alpha1/types.go#L135-L160
+
+##### PodStats CRI additions
+Previously, pod level stats were deemed the sole responsibility of the Kubelet. Currently, kubelet fetches pod level metrics by making use of cAdvisor to lookup the pod sandbox stats.
+
+
+Instead of doing so, this KEP proposes introducing a new set of CRI API endpoints, structured similarly to the `ContainerStats` endpoints, dedicated to obtaining `PodSandbox` stats.
+
+They will be defined as follows:
+
+```
+// Runtime service defines the public APIs for remote pod runtimes
+service RuntimeService {
+    ...
+    // PodSandboxStats returns stats of the pod. If the pod does not
+    // exist, the call returns an error.
+    rpc PodSandboxStats(PodSandboxStatsRequest) returns (PodSandboxStatsResponse) {}
+    // ListPodSandboxStats returns stats of all running pods.
+    rpc ListPodSandboxStats(ListPodSandboxStatsRequest) returns (ListPodSandboxStatsResponse) {}
+    ...
+}
+...
+
+message PodSandboxStatsRequest{
+    // ID of the pod for which to retrieve stats.
+    string pod_id = 1;
+}
+
+message PodSandboxStatsResponse {
+    // Stats of the pod.
+    PodSandboxStats stats = 1;
+}
+
+
+// ContainerStatsFilter is used to filter containers.
+// All those fields are combined with 'AND'
+message ContainerStatsFilter {
```

Comment:
PodSandboxStatsFilter?

Reply:
suites me right for copy pasting :upside_down_face: